common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+		for (GuideBookContents.Division div : gbc.getDivisions()) {
+			divisionsList.add(new MenuItemsDivision());
+			for (GuideBookContents.Chapter chapter : div.getChapters()) {
+				divisionsList.get(div.getDindex()).getList().add(new MenuItem(chapter));
+			}
+		}
+	}
+
+	/*
+	 * Chooses the appropriate draw methods for the current to be drawn content
+	 */
+	protected void drawCurrentEvent() {
+		int yOffset = slider.enabled ? -(int) (MathHelper.clamp(slider.getProgress() * (area$currentlyDrawnY - area$acrossY), 0, area$currentlyDrawnY - area$acrossY)) : 0;
+		area$currentlyDrawnY = state.draw(mc, mouseX, mouseY, yOffset);
+	}
+
+	/*
+	 * Calculates varius coordinates based on current width and height
+	 */
+	protected void calculateConstraints() {
+		// Start Gui
+		gui$width = (int) (6.0D / 8.0D * this.width);
+		gui$height = (int) (6.0D / 8.0D * this.height);
+		gui$innerWidth = gui$width - 2 * gui$borderWithShadowThickness;
+		gui$innerHeight = gui$height - 2 * gui$borderWithShadowThickness;
+		gui$x0 = (int) (1.0D / 8.0D * this.width);
+		gui$y0 = (int) (1.0D / 8.0D * this.height);
+		gui$x1 = gui$x0 + gui$borderWithShadowThickness;
+		gui$y1 = gui$y0 + gui$borderWithShadowThickness;
+		gui$x2 = gui$x1 + gui$innerWidth;
+		gui$y2 = gui$y1 + gui$innerHeight;
+		gui$x3 = gui$x2 + gui$borderWithShadowThickness;
+		gui$y3 = gui$y2 + gui$borderWithShadowThickness;
+		gui$bgX0 = gui$x0 + gui$borderThickness;
+		gui$bgY0 = gui$y0 + gui$borderThickness;
+		gui$bgX1 = gui$x0 + gui$width - gui$borderThickness;
+		gui$bgY1 = gui$y0 + gui$height - gui$borderThickness;
+		gui$sliderX = gui$x3 - gui$borderThickness - gui$sliderWidth;
+		gui$sliderY0 = gui$y0 + gui$borderThickness;
+		gui$sliderY1 = gui$y3 - gui$borderThickness;
+		gui$sepX0 = gui$x2 - gui$separatorThickness - gui$sliderWidth;
+		gui$sepY0 = gui$sliderY0 - gui$sepExtra;
+		gui$sepY1 = gui$sliderY0;
+		gui$sepX1 = gui$sepX0 + gui$separatorThickness + 2 * gui$shadowThickness;
+		gui$sepY2 = gui$sliderY1;
+		gui$sepY3 = gui$sliderY1 + gui$sepExtra;
+		// End Gui
+		// Calculate usable area
+		area$x0 = gui$x1;
+		area$y0 = gui$y1;
+		area$x1 = gui$x2 - 2 * gui$shadowThickness - gui$separatorThickness;
+		area$y1 = gui$y2;
+		area$acrossX = area$x1 - area$x0;
+		area$xCenter = area$x0 + area$acrossX / 2;
+		area$acrossY = area$y1 - area$y0;
+		area$offsetCenterX = (int) (1.0 / 4 * area$acrossX);
+		// End usable area
+		// Menu tiles and text
+		tile$max = area$acrossX / tile$size;
+		tile$spacing = 5;
+		while ((tile$max * (tile$size + tile$spacing)) - tile$spacing > area$acrossX) {
+			tile$max--;
+			tile$spacing = (int) ((area$acrossX % (tile$size * tile$max)) / (float) (tile$max - 1));
+		}
+		tile$spacing = (int) ((area$acrossX % (tile$size * tile$max)) / (float) (tile$max - 1));
+		// End menu
+	}
+
+	/*
+	 * Gets information from the item's nbt
+	 */
+	protected void getDataFromNBT() {
+		if (loadedNBT) return;
+		ItemStack bookItemStack = mc.player.getHeldItem(hand);
```

Comment:
From looking at this method, one shouldn't expect that the book is in the player's hand. Please pass the ItemStack as parameter and have the `mc.player.getHeldItem(hand)` in an appropriate spot

Reply:
Why wouldn't we expect the book to be in the players hand? How would he activate it otherwise?

Reply:
well, at least check if the ItemStack's item is actually the book. Can you not at all switch items while the GUI is open? What happens if there was a slight lag?